vignettes/data-in-teal.Rmd

```diff
@@ -0,0 +1,248 @@
+---
+title: "Including Data in teal Applications"
+author: "NEST CoreDev"
+output: 
+  bookdown::html_document2:
+    base_format: rmarkdown::html_vignette
+    toc: true
+    toc_depth: 5
+    number_sections: false
+vignette: >
+  %\VignetteIndexEntry{Including Data in teal Applications}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+## Data in `teal` Applications
+
+While the `teal` framework is mainly geared towards clinical data that conforms to the `ADaM` standard, general, non-relational data is handled just as well and the mechanism of passing data to applications is virtually the same.
+Also, modules in `teal.modules.general` have been designed to work with general data.
+
+All applications use the `teal_data` class as a data container.
+`teal_data` objects are passed to `init` to build the application, where modified by the filter panel (if applicable) and passed on to modules.
+Thus, the first step of building `teal` app is creating a `teal_data` object.
+
+### General Data
+
+A `teal_data` object is creating the `teal_data` function and passing data objects as `name:value` pairs.
+
+```{r, message=FALSE}
+library(teal)
+
+# create data objects
+IRIS <- iris
+CARS <- mtcars
+
+# create teal_data
+data <- teal_data(IRIS = IRIS, CARS = CARS)
```

Comment:
We used to do `IRIS = iris` just to have some non-redundant line in show-r-code (redundant: `iris <- iris`, less redundant: `IRIS <- iris`). I admit, both looks redundant.

Here we don't deal with reproducible code in `teal_data` and people might think that they really need to make all datasets uppercase and it is not the case - this limitation is only CDISC datasets because of the join keys. I think we can skip `IRIS <- iris` and specify data in `teal_data` directly to make example as much concise as possible.





```suggestion
library(teal)

# create teal_data
data <- teal_data(iris = iris, mtcars = mtcars)
```